### PR TITLE
Finished the basic code and document for v1.5 and merge it to master (fix #80)

### LINKED ISSUE #80: [DEV] ssds.pytorch v1.5 roadmap
This is the roadmap for ssds.pytorch v1.5. The development for ssds.pytorch v1.5 is fully reconstruct and almost done. The main features are listed at here:

### [Documentations](https://foreveryounggithub.github.io/ssds.doc/):
- [x] install;
- [x] basic usage;
- [x] basic parameters;
- [ ] sample tutorials;
- [x] basic python api. 

### Framework:
- [x] add training, validation, and inference support for pytorch v1.5 based model;
- [x] add support to convert the model from pytorch to onnx, and allows the user to convert the onnx model the tensorrt 7 through the code in [retinanet-examples](https://github.com/NVIDIA/retinanet-examples).

### Dataset:
- [x] remove the current the voc dataset;
- [x] add the dalicoco and dalitfrecord dataset for fast data loading.

### Anchor box matching:
- [x] remove the current anchor mathcing strategy;
- [x] add the anchor box matching for each level to make user understand ssd-like training and inference easier.

### Loss:
- [x] FocalLoss 
- [x] SmoothL1
- [x] IOU, GIOU, DIOU, CIOU Loss
- [x] MultiBoxLoss (Not recommend, not fully tested)

### Pipeline:
- [x] add DataParallel for basic multiple gpu or single gpu training (slow)
- [x] add apex for multiple gpu training (fast)

### Visualization:
- [x] add visualization for anchor strategy in each feature map (the distribution of scale and ratio in the dataset);
- [x] add visualization for defualt anchor boxes in each feature map;
- [ ] prepare the images for readme.

### Support SSDs head:
- [x] ssd;
- [x] fpn in retinanet;
- [x] bifpn in efficientdet;
- [x] yolov3 and yolov4
- [x] shelf in shelfnet

### Support backbone (feature extractor):
- [x] resnet
- [x] regnetx
- [x] mobilenet v1 and v2
- [x] shufflenet v2
- [x] darknet
- [x] densenet
- [x] efficientNet (memory cost)

### Others:
- [x] Provide the dockerfile to allow user directly build the ssds.pytorch docker quickly;
- [x] Provide the setup.py to allow user directly install the ssds.pytorch by pip;
- [ ] Prepare the pretrained models for different backbone and detection heads.

### Bug Fix:

Please let me know if you have any problem when you use the ssds.pytorch or any suggestion to make the ssds.pytorch better!

### PR DESCRIPTION
**Pull request by foreverYoungGitHub**

### CODE REVIEW

**Comment by foreverYoungGitHub** [comment]
fix #80
